react/javascript/src/components/app/SearchBar.tsx

```diff
@@ -1,85 +1,84 @@
-import React from 'react'
-import { faSearch, faQuestionCircle, faFilter } from '@fortawesome/free-solid-svg-icons'
+import React, { FunctionComponent } from 'react'
+import { faFilter, faSearch } from '@fortawesome/free-solid-svg-icons'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import SearchQueryContext from '../../SearchQueryContext'
-import statusName from '../gherkin/statusName'
-import statuses from './statuses'
+import styles from './SearchBar.module.scss'
 import { TestStepResultStatus as Status } from '@cucumber/messages'
+import statuses from './statuses'
+import statusName from '../gherkin/statusName'
 
 interface IProps {
-  statusesWithScenarios: Status[]
+  query: string
+  hideStatuses: readonly Status[]
+  statusesWithScenarios: readonly Status[]
+  onSearch: (query: string) => void
+  onFilter: (hideStatuses: Status[]) => void
 }
 
-export const SearchBar: React.FunctionComponent<IProps> = ({ statusesWithScenarios }) => {
-  const searchQueryContext = React.useContext(SearchQueryContext)
-
+export const SearchBar: FunctionComponent<IProps> = ({
+  query,
+  hideStatuses,
+  statusesWithScenarios,
+  onSearch,
+  onFilter,
+}) => {
   const searchSubmitted = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault()
     const formData = new window.FormData(event.currentTarget)
-    searchQueryContext.update({
-      query: formData.get('query').toString(),
-    })
+    onSearch(formData.get('query').toString())
   }
-
   const filterChanged = (name: Status, show: boolean) => {
-    searchQueryContext.update({
-      hideStatuses: show
-        ? searchQueryContext.hideStatuses.filter((s) => s !== name)
-        : searchQueryContext.hideStatuses.concat(name),
-    })
+    onFilter(show ? hideStatuses.filter((s) => s !== name) : hideStatuses.concat(name))
   }
-
-  const showFilters = statusesWithScenarios.length > 1
-
   return (
-    <div className="cucumber-search-bar">
-      <form className="cucumber-search-bar-search" onSubmit={searchSubmitted}>
+    <div className={styles.layout}>
+      <form className={styles.searchForm} onSubmit={searchSubmitted}>
+        <FontAwesomeIcon aria-hidden="true" className={styles.queryIcon} icon={faSearch} />
         <input
+          className={styles.queryField}
+          aria-label="Search"
           type="text"
           name="query"
-          placeholder="Some text or @tags"
-          defaultValue={searchQueryContext.query}
+          placeholder="Search with text or @tags"
+          defaultValue={query}
         />
-        <button type="submit" value="search">
-          <FontAwesomeIcon icon={faSearch} />
-        </button>
+        <small className={styles.searchHelp}>
+          You can search with plain text or{' '}
+          <a
+            href="https://cucumber.io/docs/cucumber/api/#tag-expressions"
+            target="_blank"
+            rel="noreferrer"
+          >
+            Cucumber Tag Expressions
```

Comment:
I tried `not @passing` and that highlighted `@passing`. Does it really support tag *expressions*, or just tags?

Reply:
Good question - I didn't touch any of that as part of this. I'll have a look and raise a separate issue.